File: python/lesson 6/pre_order.py
# https://leetcode.com/problems/binary-tree-preorder-traversal/

class Solution:
    def preorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        result = []
        self.preorder(root, result)
        return result
    
    def preorder(self, node: Optional[TreeNode], result: List[int]):
        if node == None: 
            return
        result.append(node.val)
        self.preorder(node.left, result)
        self.preorder(node.right, result)

File: javascript/lesson 6/preOrder.js
// https://leetcode.com/problems/binary-tree-preorder-traversal/

function preorderTraversal(root) {
    const result = [];
    preOrder(root, result);
    return result;
}

function preOrder(node, result) {
    if (!node) return;
    
    result.push(node.val);
    preOrder(node.left, result);
    preOrder(node.right, result);    
}
